aten/src/ATen/native/cuda/linalg/MagmaUtils.h

```diff
@@ -20,6 +20,7 @@ struct MAGMAQueue {
 
   // Constructor
   explicit MAGMAQueue(int64_t device_id) {
+    c10::cuda::device_synchronize();
```

Comment:
If the goal is to affect LU tests then this synchronize shall be localized to LU function only, here's the place where MAGMAQueue is created: https://github.com/pytorch/pytorch/blob/54c46e4f902209dc9cca2fbeb8181acf05409cb6/aten/src/ATen/native/cuda/linalg/BatchLinearAlgebra.cpp#L1564